planetiler-core/src/main/java/com/onthegomap/planetiler/util/FileUtils.java

```diff
@@ -314,4 +322,29 @@ public static void unzip(InputStream input, Path destDir) {
       throw new UncheckedIOException(e);
     }
   }
+
+  /**
+   * Unzips files from {@param zipPath} matching the glob {@param pattern} to {@param tmpLocation}. Extracted files will
+   * be deleted when the JVM exits.
+   *
+   * @return list of paths to extracted files
+   */
+  public static List<Path> unzipMatchingFiles(Path zipPath, String pattern, Path tmpLocation) throws IOException {
+    PathMatcher matcher = createGlobMatcher(zipPath.getFileSystem(), pattern);
+
+    try (var zipFs = FileSystems.newFileSystem(zipPath)) {
+      return walkFileSystemWithMatcher(zipFs, matcher)
+        .map(path -> {
+          try {
+            Path unzippedPath = tmpLocation.resolve(path.getFileName().toString());
```

Comment:
Probably unlikely but there could be 2 files with the same name in different directories with the same name they will clobber eachother.

Also, I think `resolve` can take a path so you don't need toString